src/theory/builtin/proof_checker.h

```diff
@@ -73,6 +73,17 @@ class BuiltinProofRuleChecker : public ProofRuleChecker
    * @return The rewritten form of n.
    */
   static Node applyRewrite(Node n, MethodId idr = MethodId::RW_REWRITE);
+  /**
+   * Apply small-step rewrite on n (either pre- or post-rewrite). This
+   * encapsulates the exact behavior of a THEORY_REWRITE step in a proof.
+   * Rewriting is performed on the Skolem form of n.
+   *
+   * @param n The node (in witness form) to rewrite
```

Comment:
This actually needs to be Skolem form, since it could in theory involve updates to the witness term that destroy its structure.